mlflow/pyspark/ml/__init__.py

```diff
@@ -999,7 +1010,29 @@ def _infer_model_signature(input_example_slice):
                     model_output = spark_model.transform(input_slice_df).drop(
                         *input_slice_df.columns
                     )
-                    return infer_signature(input_example_slice, model_output.toPandas())
+                    # TODO: Remove this once we support non-scalar spark data types
+                    unsupported_columns = _get_columns_with_unsupported_data_type(model_output)
+                    if unsupported_columns:
+                        _logger.warning(
+                            "Model output contains unsupported spark data types: "
+                            f"{unsupported_columns}. Output signature will not be logged."
+                        )
+                        model_output = None
+                    else:
+                        model_output = model_output.toPandas()
+
+                    return infer_signature(input_example_slice, model_output)
+
+                # TODO: Remove this once we support non-scalar spark data types
+                nonlocal log_model_signatures
+                if log_model_signatures:
+                    unsupported_columns = _get_columns_with_unsupported_data_type(input_df)
+                    if unsupported_columns:
+                        _logger.warning(
+                            "Model input contains unsupported spark data types: "
+                            f"{unsupported_columns}. Model signature will not be logged."
+                        )
+                        log_model_signatures = False
```

Comment:
This line it disables `log_model_signatures`, then seems we don't need changes in L1014-L1020 ?

Reply:
The model output might contain non-scalar columns, right?

Reply:
ok, then it makes sense.